templates/home/jenkins/init.groovy.d/puppet.bootstapping.groovy.erb

```diff
@@ -0,0 +1,92 @@
+#!groovy
+/*
+
+  !! Managed by Puppet !!
+  Setup initial bootstrapping user for puppet
+
+*/
+
+import jenkins.model.*
+import hudson.security.*
+import com.cloudbees.plugins.credentials.*
+import hudson.plugins.active_directory.*
+import hudson.security.HudsonPrivateSecurityRealm.Details
+
+// Code
+def instance = Jenkins.getInstance()
+def currentRealm = instance.getSecurityRealm().getClass().getName()
+
+// bootstrap users
+def bootstrapUsers = [
+<% @_bootstrap_users.each do |username,userconfig| -%>
+  [
+    username : '<%= username %>',
+    password : '<%= userconfig['password'] %>',
+    email : '<%= userconfig['email'] %>',
+    full_name : '<%= userconfig['full_name'] %>',
+    ssh_key : '<%= userconfig['public_key'] %>',
+  ],
+<% end -%>
+]
+
+if (currentRealm == 'hudson.security.HudsonPrivateSecurityRealm') {
```

Comment:
If there is a different realm configured, say LDAP, this test will fail and will overwrite the ldap configuration. Shouldn't it be something like `if (currentRealm != 'hudson.security.SecurityRealm$None') {`, more or less?